src/update.c

```diff
@@ -317,36 +328,50 @@ int main_update()
 
 	read_subscriptions_alt(&current_subs);
 
-	/* Step 1: get versions */
-
+/* Step 1: get versions */
+version_check:
 	ret = check_versions(&current_version, &server_version, path_prefix);
 
 	if (ret < 0) {
 		ret = EXIT_FAILURE;
 		goto clean_curl;
 	}
-	if (server_version <= current_version) {
-		fprintf(stderr, "Version on server (%i) is not newer than system version (%i)\n", server_version, current_version);
-		ret = EXIT_SUCCESS;
-		goto clean_curl;
-	}
 
 	if (mix_exists) {
 		ret = check_mix_versions(&mix_current_version, &mix_server_version, path_prefix);
 		if (ret < 0) {
 			ret = EXIT_FAILURE;
 			goto clean_curl;
 		}
-		ret = check_manifests_uniqueness(server_version, mix_server_version);
-		if (ret) {
-			printf("\n\t!! %i collisions were found between mix and upstream, please re-create mix !!\n", ret);
-		} else {
-			printf("OK! Manifests are compatible\n");
+		/* Check if a new upstream version is available so we can update to it still */
+		if (need_new_upstream(server_version)) {
+			printf("NEW CLEAR AVAILABLE %d\n", server_version);
+			ret = check_manifests_uniqueness(server_version, mix_server_version);
+			if (ret) {
+				printf("\n\t!! %i collisions were found between mix and upstream, please re-create mix !!\n", ret);
+			}
+
+			/* Update the clearversion that will be used to generate the new mix content */
+			FILE *verfile = fopen(MIX_DIR ".clearversion", "w+");
+			if (!verfile) {
+				fprintf(stderr, "ERROR: fopen() returned %s\n", strerror(errno));
```

Comment:
That's not guaranteed, that'd be undefined in this case depending on why fopen failed on that file.What do you recommend instead?